src/core/detail/bptree_internal.h

```diff
@@ -370,6 +370,14 @@ template <typename Comp>
 auto BPTreeNode<T>::BSearch(KeyT key, Comp&& cmp_op) const -> SearchResult {
   uint16_t lo = 0;
   uint16_t hi = num_items_;
+  assert(hi > 0);
+
+  // optimization: check the last item first.
```

Comment:
Questionable. Why not the first? Is it some property of B+ trees?

 Your binsearch first checks the middle element

Reply:
The reason for this is how we add multiple entries to zset. We sort them before and if this was an empty set we save some cycles.

Reply:
Ah, its from #1885 👍🏻

Reply:
Btw, then move the right boundary of the binary search to h - 1

Reply:
i tried,. it actually incorrect because it is possible that key is equal to last item. but thinking about it now I can handle this as well.

Reply:
But if cmp == 0 its equal? So you need to check >= 0, and found = cmp == 0

Reply:
done, PTAL